client/src/main/java/org/apache/cassandra/sidecar/client/RequestExecutor.java

```diff
@@ -101,10 +100,20 @@ public <T> T executeRequest(RequestContext context, long timeout, TimeUnit unit)
      */
     public <T> CompletableFuture<T> executeRequestAsync(RequestContext context)
     {
+        Iterator<SidecarInstance> iterator = context.instanceSelectionPolicy().iterator();
+        SidecarInstance instance = iterator.next();
```

Comment:
This seems wrong. `iterator.next` if implemented according to spec, will throw a `NoSuchElementException` if the iteration has no more elements. We must call `hasNext` before calling `next`

Reply:
good call.